src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs

```diff
@@ -70,7 +70,7 @@ public InteractiveRunner(ScriptCompiler scriptCompiler, ScriptLogger logger, Scr
                 }
                 else
                 {
-                    var lineDependencies = ScriptCompiler.RuntimeDependencyResolver.GetDependenciesFromCode(CurrentDirectory, input);
+                    var lineDependencies = ScriptCompiler.RuntimeDependencyResolver.GetDependenciesFromCode(CurrentDirectory, input).SelectMany(rtd => rtd.Assemblies).Distinct();
```

Comment:
this hypothetical `WithScriptMap` against the resolver inside `ScriptOptions` should be invoked from here